spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDelete.scala

```diff
@@ -77,7 +61,7 @@ object RewriteDelete extends Rule[LogicalPlan] with PredicateHelper with Logging
       val writeInfo = newWriteInfo(r.schema)
       val mergeBuilder = r.table.asMergeable.newMergeBuilder("delete", writeInfo)
 
-      val scanPlan = buildScanPlan(r.table, r.output, mergeBuilder, cond)
+      val scanPlan = buildScanPlan(spark, r.table, r.output, mergeBuilder, Some(cond))
```

Comment:
I would prefer to keep changes for MERGE INTO separate from DELETE FROM right now. That will keep PRs smaller and we can focus on getting MERGE INTO right, without the distraction of sometimes breaking DELETE FROM. Once we are happy with it, then we can refactor to produce common utilities.

Reply:
@rdblue Okay Ryan.